pageserver/src/object_repository.rs

```diff
@@ -788,6 +780,28 @@ impl Timeline for ObjectTimeline {
                                 self.obj_store.unlink(&key, lsn)?;
                                 result.deleted += 1;
                             }
+                        } else if let RelishTag::TwoPhase { xid: _ } = rel {
+                            // Reconstruct TwoPhase file at horizon unless it was dropped
```

Comment:
Do we need this? Doesn't the else-path below with comment `versioned always materialized objects...` handle this just fine?

Reply:
You're right. I was confused by the fact that we store WAL records for such objects.

I wonder if it would be better to always materialize such WAL records in `save_decoded_record` by using `put_page_image()` instead of `put_wal_record()` ?

Reply:
Makes sense